dask/dataframe/core.py

```diff
@@ -1683,6 +1687,12 @@ def mean(self, axis=None, skipna=True, split_every=False, dtype=None, out=None):
             )
             if isinstance(self, DataFrame):
                 result.divisions = (min(self.columns), max(self.columns))
+            if dtype == "<M8[ns]":
+                val = pd.Timestamp(result.compute())
```

Comment:
We don't want to `compute` here. We should only compute when the user requests it. You should be able to return a delayed result.